source/eitri/docker_wrapper.py

```diff
@@ -0,0 +1,107 @@
+#
+# Hope you'll enjoy and contribute to this project,
+#     powered by Canal TP (www.canaltp.fr).
+# Help us simplify mobility and open public transport:
+#     a non ending quest to the responsive locomotion way of traveling!
+#
+# LICENCE: This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Stay tuned using
+# twitter @navitia
+# IRC #navitia on freenode
+# https://groups.google.com/d/forum/navitia
+# www.navitia.io
+
+import docker
+import logging
+import retrying
+import psycopg2
+
+# postgres/postgis image
+POSTGIS_IMAGE = 'github.com/helmi03/docker-postgis.git'
+POSTGIS_CONTAINER_NAME = 'postgis:2.1'
+
+
+class DbParams(object):
+    def __init__(self, host, dbname, user, password):
+        self.host = host
+        self.user = user
+        self.dbname = dbname
+        self.password = password
+
+    def cnx_string(self):
+        return 'postgresql://{u}:{pwd}@{h}/{dbname}'.format(h=self.host,
+                                                            u=self.user,
+                                                            dbname=self.dbname,
+                                                            pwd=self.password)
+
+
+class PostgresDocker(object):
+    """
+    launch a temporary docker with a postgresql db
+    """
+    def __init__(self):
+        log = logging.getLogger(__name__)
+        self.docker = docker.Client(base_url='unix://var/run/docker.sock')
+
+        log.info("building the temporary docker image")
+        self.docker.build(POSTGIS_IMAGE, tag=POSTGIS_CONTAINER_NAME, rm=True)
+
+        self.container_id = self.docker.create_container(POSTGIS_CONTAINER_NAME).get('Id')
+
+        log.info("docker id is {}".format(self.container_id))
+
+        log.info("starting the temporary docker")
+        self.docker.start(self.container_id)
+        self.ip_addr = self.docker.inspect_container(self.container_id).get('NetworkSettings', {}).get('IPAddress')
+
+        if not self.ip_addr:
+            log.error("temporary docker {} not started".format(self.container_id))
+            exit(1)
+
+        # we poll to ensure that the db is ready
+        def db_cnx():
+            psycopg2.connect(self.get_db_params().cnx_string())
+
+        retrying.Retrying(stop_max_delay=10000,
+                          wait_fixed=100,
+                          retry_on_exception=lambda e: isinstance(e, Exception)).call(db_cnx)
+
```

Comment:
Minor: won't it be nicer if you do: 

```Python
@retry(stop_max_delay=10000, wait_fixed=100, 
             retry_on_exception=lambda e: isinstance(e, Exception))
 def db_cnx(self):
    psycopg2.connect(self.get_db_params().cnx_string())
```
then you call `self.db_cnx()` in `__init__`